pkg/record/recorder.go

```diff
@@ -65,7 +49,7 @@ func Event(object runtime.Object, reason, message string) {
 
 // Eventf is just like Event, but with Sprintf for the message field.
 func Eventf(object runtime.Object, reason, message string, args ...interface{}) {
-	defaultRecorder.Eventf(object, corev1.EventTypeNormal, strings.Title(reason), message, args)
```

Comment:
Since there is a defaultRecorder, that implies to me there could be a customized recorder. This is a very similar pattern to the http client that the net/http package exposes, but, like the net/http package, we should also export our DefaultRecorder so people can use methods on it if they don't care.

Reply:
Users can customize the internal recorder through the Init method offered by this package. I usually try to avoid exposing too much unless strictly necessary. Which seems fine in this case 😃